dotnet/samples/KernelSyntaxExamples/Example57_KernelHooks.cs

```diff
@@ -184,7 +184,7 @@ private static async Task BeforeInvokeCancellationAsync()
         // Adding new inline handler to cancel/prevent function execution
         kernel.FunctionInvoking += (object? sender, FunctionInvokingEventArgs e) =>
         {
-            Console.WriteLine($"{e.Function.GetMetadata().PluginName}.{e.Function.GetMetadata().Name} : FunctionInvoking - Cancelling all subsequent invocations");
+            Console.WriteLine($"{e.Function.Metadata.PluginName}.{e.Function.Metadata.Name} : FunctionInvoking - Cancelling all subsequent invocations");
```

Comment:
Why metadata has Plugin Name and root doesn't?
I believe we should remove this property from Metadata... 
Same for above...
```suggestion
            Console.WriteLine($"{e.Function.Metadata.PluginName}.{e.Function.Name} : FunctionInvoking - Cancelling all subsequent invocations");
```